engine/src/main/java/org/terasology/engine/world/block/BlockArea.java

```diff
@@ -354,4 +355,24 @@ public int hashCode() {
     public String toString() {
         return "BlockArea[(" + this.minX + ", " + this.minY + ")...(" + this.maxX + ", " + this.maxY + ")]";
     }
+
+    @Override
+    public long distanceSquared(Vector2ic p){
+        return distanceSquared(p.x(), p.y());
+    }
+
+    @Override
+    public long distanceSquared(int px, int py){
+        //center = min + width / 2 = min + (max - min) / 2
+        // 2 * center  = 2 * min + max - min = min + max
+        int cx2 = minX() + maxX();
+        int cy2 = minY() + maxY();
+
+        //     dx = p - center - width / 2
+        // 2 * dx = 2 * px - 2 * center - width
+        //     dx = (2 * px - 2 * center - width) / 2
+        long dx = Math.max(Math.abs(2 * px - cx2) - getSizeX(), 0) / 2;
+        long dy = Math.max(Math.abs(2 * py - cy2) - getSizeY(), 0) / 2;
+        return dx * dx + dy * dy;
```

Comment:
could we move these to the edges of the bounding box. maybe it would be more useful to use floating point instead of int px and int py. just need that change and I will approve this.

Reply:
I did create a test version for float valued arguments and along with changing the return type, there is a considerable difference in the values returned by it. The code is ready to be pushed along with changed test values but would it not affect the results where this function is used? The benefit to calculating distance this way is that this function returns 0 for all the values touching the block.

Reply:
min[X,Y,Z](), max[X,Y,Z]() are the center of blocks. but the actual boundaries are -.5 and +.5. umm can't look at the code right now to point at an example but there should be some code that shows this behaviour.